### DISCUSSION TITLE
Should applease blob container be empty when enabled?

### DISCUSSION THREAD

**Discussion by mpaul31** [Azure/azure-functions-durable-extension#1812] [Q&A] [answered]
I know this is a relatively new feature and I checked the traces and shows it is enabled, however, the `applease` blob container is always empty. Is this expected? My hunch says it should not but wanted to double-check. Might also be worth pointing out I am overriding the TaskHubName in the host.json.

**Comment by cgillum** [comment]
Pinging @amdeel for this question.

**Comment by amdeel** [comment, marked as answer]
This is expected. The container itself is what is currently being leased with the app lease feature. Because there hadn't been a need to store any information in our initial iteration of this feature, it wasn't necessary to actually add a blob to the container, just use it to represent the lease.

In an [upcoming update](https://github.com/Azure/azure-functions-durable-extension/issues/1503) to the app lease feature, there will be a blob inside that container that stores information about which app currently owns the lease. This update is really cool because it will allow apps to steal the app lease using a new http api and will also let apps automatically reset themselves to the waiting stage, trying to acquire the app lease for failure situations again. Previously, swapping apps and resetting them required manual restarts to applications.

**Comment by mpaul31** [comment]
@amdeel One of our primary apps is currently exceeding the 32 character host id limit (we get a warning in diagnose and solve blade) and we have not been able to delete the resource and re-deploy yet.

> Currently when generating a default host ID we use the host name (slot host name) and we would truncate it to 32 characters max. This ensures that the generated ID conforms to the core SDK length restrictions.
> 
> This truncation can open the possibility for naming collisions, particularly in the case of slots. In slot scenarios, if the site name is over 32 characters long and a slot is created that starts with the same 32 characters and is only disambiguated in later characters, both the production and the slot site will be using the same host ID.

Will this cause any issues with the new app lease functionality? I looked at the blob inside the container and it looks to be a GUID so I'm not 100% sure or not.

Thanks!
